File: onyx-database-tests/src/main/java/entities/schema/SchemaIdentifierChangedEntity.java
package entities.schema;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.ManagedEntity;
import com.onyx.persistence.annotations.Attribute;
import com.onyx.persistence.annotations.Entity;
import com.onyx.persistence.annotations.Identifier;
import com.onyx.persistence.annotations.values.IdentifierGenerator;

/**
 * Created by tosborn1 on 8/26/15.
 */

@Entity
public class SchemaIdentifierChangedEntity extends ManagedEntity implements IManagedEntity
{
    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    public long id;

    @Attribute(nullable=true)
    public Long longValue;
}

File: onyx-database-tests/src/main/java/entities/schema/SchemaIdentifierChangedEntity.kt
package entities.schema

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.ManagedEntity
import com.onyx.persistence.annotations.Attribute
import com.onyx.persistence.annotations.Entity
import com.onyx.persistence.annotations.Identifier
import com.onyx.persistence.annotations.values.IdentifierGenerator

/**
 * Created by Tim Osborn on 8/26/15.
 */

@Entity
class SchemaIdentifierChangedEntity : ManagedEntity(), IManagedEntity {
    @Identifier(generator = IdentifierGenerator.SEQUENCE)
    @Attribute
    var id: Long = 0

    @Attribute(nullable = true)
    var longValue: Long? = null
}